WooCommerce/Classes/ViewRelated/Dashboard/Settings/CardReaders/CardReaderSettingsViewModel.swift

```diff
@@ -62,80 +103,84 @@ final class CardReaderSettingsViewModel: ObservableObject {
-            //print("==== new card readers discovered ends   ====")
+        let action = CardPresentPaymentAction.startCardReaderDiscovery(siteID: siteID, onCompletion: { [weak self] cardReaders in
+            self?.didDiscoverReaders(cardReaders: cardReaders)
         })
 
-        ServiceLocator.stores.dispatch(discoveryAction)
-
-        let discoveryCancellationAction = CardPresentPaymentAction.cancelCardReaderDiscovery { status in
-            // TODO. To be implemented
-            // Leaving these prints here to help test the PR
-            print("===== View model - cancellation callback ====")
-            print("View model. new status received ", status)
-            //print("===== View model - cancellation ends ====")
-        }
-
-        ServiceLocator.stores.dispatch(discoveryCancellationAction)
-
-        // TODO Remove - simulates searching with a timer
-        //timer = Timer.scheduledTimer(timeInterval: 3, target: self, selector: #selector(dummyFoundReader), userInfo: nil, repeats: false)
+        ServiceLocator.stores.dispatch(action)
     }
 
-    // TODO Remove
-    @objc private func dummyFoundReader() {
-        foundReader = CardReader(serialNumber: "CHB204909001234", batteryLevel: 0.885)
-        activeAlert = .foundReader
-    }
+    /// Called when a reader has been discovered (with an array of discovered readers).
+    private func didDiscoverReaders(cardReaders: [CardReader]) -> Void {
+        // TODO - more than one reader? sort in the manner in which we'd like the rows presented
+        guard activeAlert != .foundReader else {
+            return
+        }
 
-    // TODO Remove
-    @objc private func dummyUpdateBattery() {
-        guard var batteryLevel = connectedReader?.batteryLevel else {
+        guard cardReaders.count > 0 else {
             return
         }
 
-        batteryLevel = batteryLevel * 0.99
-        connectedReader?.batteryLevel = batteryLevel
+        // TODO - show the multiple-readers-found UITableView when more than one reader is found
+        // For now, just show the tail (last) of the array
+        guard let cardReader = cardReaders.last else {
+            return
+        }
+        foundReaders = [cardReader]
+        activeAlert = .foundReader
     }
 
+    /// Views can call this method to stop searching for readers.
     func stopSearch() {
         // TODO dispatch an action to stop searching.
-        timer?.invalidate() // TODO Remove
         activeAlert = .none
     }
 
+    /// Views can call this method to connect to a found reader.
     func connect() {
-        // TODO dispatch an action to connect.
         activeAlert = .connecting
 
-        // TODO Remove - simulates connecting with a timer
-        timer = Timer.scheduledTimer(timeInterval: 3, target: self, selector: #selector(dummyConnectedReader), userInfo: nil, repeats: false)
+        let action = CardPresentPaymentAction.connect(reader: foundReaders[0], onCompletion: { [weak self] result in
```

Comment:
When we handle more than one found reader, we'll need to update this. We'll need to decide what to pass to connect to "pick" the right reader.